pkg/core/hash.go

```diff
@@ -41,16 +41,36 @@ func calculateConfigHash(children []Object) (string, error) {
 	// Add the data from each child to the hashSource
 	// All children should be in the same namespace so each one should have a
 	// unique name
-	for _, obj := range children {
-		switch child := obj.(type) {
-		case *corev1.ConfigMap:
-			cm := corev1.ConfigMap(*child)
-			hashSource.ConfigMaps[child.GetName()] = cm.Data
-		case *corev1.Secret:
-			s := corev1.Secret(*child)
-			hashSource.Secrets[child.GetName()] = s.Data
-		default:
-			return "", fmt.Errorf("passed unknown type: %v", reflect.TypeOf(child))
+	for _, child := range children {
+		if child.object != nil {
+			switch childObject := child.object.(type) {
+			case *corev1.ConfigMap:
+				cm := corev1.ConfigMap(*childObject)
```

Comment:
These two cases got big, can we just factor them out into separate methods somehow so the switch can look more like: 

```
switch child.object.(type){
case *corev1.ConfigMap:
  hashSource.ConfigMaps[childObject.GetName()] = getConfigMapData(child)
case *corev1.Secret:
  hashSource.Secrets[childObject.GetName()] = getSecretData(child)
default:
  return "", fmt.Errorf("passed unknown type: %v", reflect.TypeOf(child))
}
```

Reply:
Done.  By the time we get into the new methods, golang can't assert the types any more, so there has to be a type assertion.  i.e.

```golang
func getConfigMapData(child configObject) map[string]string {
  cm := corev1.ConfigMap(*child.object.(*corev1.ConfigMap)) // Casting with a type assertion 
  ...
```